sklearn/linear_model/logistic.py

```diff
@@ -1281,9 +1287,9 @@ def fit(self, X, y, sample_weight=None):
         self.n_iter_ = np.asarray(n_iter_, dtype=np.int32)[:, 0]
 
         if self.multi_class == 'multinomial':
-            self.coef_ = fold_coefs_[0][0]
+            self.coef_ = fold_coefs_[0][0].astype(_dtype)
         else:
-            self.coef_ = np.asarray(fold_coefs_)
+            self.coef_ = np.asarray(fold_coefs_, dtype=_dtype)
```

Comment:
Is the conversion necessary here? In other word, if we get the code right, doesn't coefs_ get returned in the right dtype?